SteamReview/SteamReview/Steam.cpp

```diff
@@ -0,0 +1,320 @@
+	if (std::cin.peek() == '\n')
+	{
+		ActiveMenuCommand = EMenuCommand::MainMenu;
+	}
+};
+
+//------------------------------------------------------
+void FSteam::AppRun()
+{
+	while (ActiveMenuCommand != EMenuCommand::Exit)
+	{
+		switch (ActiveMenuCommand)
+		{
+		case EMenuCommand::MainMenu:
+			OpenMainMenu();
+			break;
+		case EMenuCommand::AddGame:
+			OpenAddGame();
+			break;
+		case EMenuCommand::CategoryManager:
+			OpenCategoryManager();
+			break;
+		case EMenuCommand::DisplayGames:
+			OpenDisplayGames();
+			break;
+		case EMenuCommand::Exit:
+			break;
+		}
+	}
+}
+
+//---------------------------------------------------------------
+int FSteam::InputValidate(int InputValue, const int IntLowValueToValid, const int IntHighValueToValid)
+{
+	while (std::cin.fail() || (InputValue < IntLowValueToValid) || (InputValue > IntHighValueToValid))
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid Option, please enter a valid Option " << std::endl;
+		std::cin >> InputValue;
+	}
+	return InputValue;
+};
+
+//--------------------------------------------------------------
+FGame FSteam::AddANewGame()
+{
+	std::system("cls");
+	std::cout << "Let's add a new game!" << std::endl;
+	std::cout << "-----------------------------------" << std::endl;
+
+	//adding the game
+	std::cin.ignore(500, '\n');
+	std::cout << "Please enter the game's name" << std::endl;
+	std::string InGameName;
+	std::getline(std::cin, InGameName);
+
+	std::cout << "Please enter the name of the game's developer studio" << std::endl;
+	std::string InStudioName;
+	std::getline(std::cin, InStudioName);
+
+	std::cout << "------------------------------------" << std::endl;
+	std::cout << "Please introduce the release date" << std::endl;
+	std::cout << "------------------------------------" << std::endl;
+
+	std::cout << "Please enter Day ";
+	int InDay;
+	std::cin >> InDay;
+	InputValidate(InDay, 0, 31);
+
+	std::cout << "Please enter Month ";
+	int InMonth;
+	std::cin >> InMonth;
+	InputValidate(InMonth, 0, 12);
+
+	std::cout << "Please enter Year ";
+	int InYear;
+	std::cin >> InYear;
+	InputValidate(InYear, 1990, 2022);
+
+	//Creating a Game
+	return FGame(InGameName, InStudioName, InDay, InMonth, InYear);
+
+
+};
+
+//----------------------------------------------------------------
+bool FSteam::AddNewGameToCategory(const FGame InGame, const int UserInput)
+{
+	InputValidate(UserInput, 1, 2);
+	if (UserInput == 1)
+	{
+		//Choosing a category where the user going to add a game
+		std::system("cls");
+		std::cout << "------------------------------------------\nPlease Select the category where you want to add a game" << std::endl;
+		CategoryContainer.PrintCategories();
+
+		int CategoryChosen;
+		std::cin >> CategoryChosen;
+		InputValidate(CategoryChosen, 0, CategoryContainer.GetCurrentCategoryAmount());
+
+		//adding the game to category
+		if (CategoryContainer.AddGameToCategory(CategoryChosen - 1, InGame))
```

Comment:
Careful here. You're allowing the input to be 0, but you're using the input - 1. That means that if you choose 0, you'd be giving an input of -1.